fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanSummary.java

```diff
@@ -211,10 +211,11 @@ public void zeroFields() {
 
     public void updateSummary(final MonetaryCurrency currency, final Money principal,
             final List<LoanRepaymentScheduleInstallment> repaymentScheduleInstallments, final LoanSummaryWrapper summaryWrapper,
-            final Boolean disbursed, Set<LoanCharge> charges) {
+            final Boolean disbursed, Set<LoanCharge> charges, final Money chargeback) {
 
         this.totalPrincipalDisbursed = principal.getAmount();
-        this.totalPrincipalRepaid = summaryWrapper.calculateTotalPrincipalRepaid(repaymentScheduleInstallments, currency).getAmount();
+        this.totalPrincipalRepaid = summaryWrapper.calculateTotalPrincipalRepaid(repaymentScheduleInstallments, currency).minus(chargeback)
```

Comment:
Are you sure we need this?
I dont think we need to adjust the repaid principal amount

Reply:
We need It because If we don't consider the charge back transaction, the summary is not updated properly the principal part and It results in a wrong outstanding balance, so with that we'll have 

PrincipalOutstanding = Principal - Repayment(s) - Waive/WritenOff* + Charge back(s)*

Where: * If exists some transaction of these types